pkg/provisioner/ironic/ironic.go

```diff
@@ -472,9 +503,11 @@ func (p *ironicProvisioner) ValidateManagementAccess(data provisioner.Management
 		}
 
 		// The actual password is not returned from ironic, so we want to
-		// update the whole DriverInfo only if the credentials have changed
-		// otherwise we will be writing on every call to this function.
-		if credentialsChanged {
+		// update the whole DriverInfo only if the credentials or BMC address
+		// has changed, otherwise we will be writing on every call to this
+		// function.
+		if credentialsChanged || bmcAddressChanged {
+			p.log.Info("Updating driver info because the credentials and/or the BMC address changed")
 			updater.SetTopLevelOpt("driver_info", driverInfo, ironicNode.DriverInfo)
```

Comment:
This is calling the PATCH API to set the node's driver_info field to the value of `driverInfo` variable. Which does not seem to contain your changes (you're updating `ironicNode.DriverInfo`).

Reply:
The changes are already present in the driverInfo variable, like you noted above.